sqllogictest-bin/src/main.rs

```diff
@@ -666,8 +666,29 @@ impl RunResult {
     }
 }
 
+trait Output: Write {
+    fn finish(&mut self) -> io::Result<()>;
+}
+
+/// In serial mode, we directly write to stdout.
+impl Output for Stdout {
+    fn finish(&mut self) -> io::Result<()> {
+        self.flush()
+    }
+}
+
+/// In parallel mode, we write to a buffer and flush it to stdout at the end
+/// to avoid interleaving output from different parallelism.
+impl Output for Vec<u8> {
+    fn finish(&mut self) -> io::Result<()> {
+        let mut stdout = stdout();
```

Comment:
This does not guarantee the exclusiveness of the handler?

The std doc says,

> Each handle returned is a reference to a shared global buffer whose access is synchronized via a mutex. If you need more explicit control over locking, see the [Stdout::lock](https://doc.rust-lang.org/stable/std/io/struct.Stdout.html#method.lock) method.

Reply:
Good catch. However, it appears that `write_all` is not interruptable and we only call it once, so we still won't observe interleaving output. Nonetheless, `lock` seems always better.